lib/ddtrace/sampler.rb

```diff
@@ -0,0 +1,48 @@
+
+module Datadog
+  # \Sampler performs client-side trace sampling.
+  class Sampler
+    def sample(span); end
+  end
+
+  # \AllSampler samples all the traces.
+  class AllSampler < Sampler
+    def sample(span)
+      span.sampled = true
+    end
+  end
+
+  # \RateSampler is based on a sample rate.
+  class RateSampler < Sampler
+    KNUTH_FACTOR = 1111111111111111111
+    SAMPLE_RATE_METRIC_KEY = '_sample_rate'.freeze()
+
+    attr_reader :sample_rate
+
+    # Initialize a \RateSampler.
+    # This sampler keeps a random subset of the traces. Its main purpose is to
+    # reduce the instrumentation footprint.
+    #
+    # * +sample_rate+: the sample rate as a \Float between 0.0 and 1.0. 0.0
+    #   means that no trace will be sampled; 1.0 means that all traces will be
+    #   sampled.
+    def initialize(sample_rate = 1.0)
+      unless sample_rate > 0.0 && sample_rate <= 1.0
+        raise ArgumentError,
```

Comment:
While I agree with raising an error in the context of an important/business piece of code, we should shoot for something more conservative as an monitoring/instrumentation lib.

The same way as we don't raise exceptions if the Agent endpoint is wrong, if one does something wrong with a span, ... we should behave as a 0-footprint best-effort lib.

And in general, as a user I'd be worried if my APM solution can break my app.